File: python/dnd-character/dnd_character.py
from random import randint
from math import floor

def modifier(constitution):
    return floor((constitution - 10) / 2)

class Character:
    def __init__(self):
        self.strength = randint(1, 6) + randint(1, 6) + randint(1, 6)
        self.dexterity = randint(1, 6) + randint(1, 6) + randint(1, 6)
        self.constitution = randint(1, 6) + randint(1, 6) + randint(1, 6)
        self.intelligence = randint(1, 6) + randint(1, 6) + randint(1, 6)
        self.wisdom = randint(1, 6) + randint(1, 6) + randint(1, 6)
        self.charisma = randint(1, 6) + randint(1, 6) + randint(1, 6)
        self.hitpoints = 10 + modifier(self.constitution)

    def ability(self):
        return self.strength

File: kotlin/dnd-character/src/main/kotlin/DndCharacter.kt
import kotlin.math.*
import kotlin.collections.slice

class DndCharacter {

    val strength: Int = ability()
    val dexterity: Int = ability()
    val constitution: Int = ability()
    val intelligence: Int = ability()
    val wisdom: Int = ability()
    val charisma: Int = ability()
    val hitpoints: Int = 10 + modifier(constitution)

    companion object {

        fun ability() = List(4) { (1..6).random() }
            .sorted()
            .slice(1..3)
            .sum()

        fun modifier(score: Int) = floor((score - 10) / 2.0).toInt()
    }

}
